webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/InterpolatedValueCompiler.scala

```diff
@@ -236,6 +239,7 @@ trait AnalyseInterpolation[T, I <: GenericInterpolationContext[T]] {
       case NonRudderVar(s)     => Right(s"$${${s}}")
       case NodeAccessor(path)  => getNodeAccessorTarget(context, path)
       case Param(path)         => getRudderGlobalParam(context, path)
+      case RudderEngine(e, m, p) => Right("encrypted-password-toto") // TODO remove only testing purpose replace by match case and engine's logic
```

Comment:
you will need to create a service that is able to look-up engines and pass them the method and param, ie something like:

```
trait RudderPropertyEngine { def name: String, def process(namespace: List[String], parameters: ConfigValue): IOResult[String] } 
class RudderPropertyEngineService(listOfEngine: List[RudderPropertyEngine]) {
  val engines = listOfEngine.map(e => (e.name.toLowerCase, e)

  def process(engine: String, namespace: List[String], param: ConfigValue): IOResult[String] = engines.get(engine.toLowerCase).map(e => e.process(namespace, param)).notOptionnal(s"Engine '${engine}' not found. Parameter can not be expanded: ${param}")
}
```

And then, for test, you can implement a false encrypted password engine with mock values.